inventory.py

```diff
@@ -93,57 +99,109 @@ def gather_course_info_for_account(account_id: int, term_id: int) -> Sequence[in
     users_string = ','.join([str(user_id) for user_id in user_ids])
-
     user_query = f'''
-        SELECT u.id AS user_id,
-               u.canvas_id AS user_canvas_id,
-               u.name AS user_name,
-               u.workflow_state AS user_workflow_state,
-               p.unique_name AS pseudonym_uniqname
+        SELECT u.id AS warehouse_id,
+               u.canvas_id AS canvas_id,
+               u.name AS name,
+               p.unique_name AS uniqname,
+               u.workflow_state AS workflow_state
         FROM user_dim u
         JOIN pseudonym_dim p
             ON u.id=p.user_id
         WHERE u.id in ({users_string});
     '''
-
     logger.info('Making user_dim query')
     user_df = pd.read_sql(user_query, UDW_CONN)
-    user_df.to_csv(os.path.join('data', 'user.csv'), index=False)
-    logger.info('User data was written to data/user.csv')
+    # Found that the IDs are not necessarily unique, so dropping duplicates
+    user_df = user_df.drop_duplicates(subset=['warehouse_id', 'canvas_id'])
+    logger.debug(user_df.head())
+    return user_df
+
+
+def check_if_valid_user_id(id: int, user_ids) -> bool:
+    if id in user_ids:
+        return True
+    else:
+        return False
+
+
+def run_course_inventory() -> None:
+    start = datetime.now()
+
+    # Gather course data
+    course_df = gather_course_info_for_account(1, ENV['TERM_ID'])
+
+    # Gather enrollment data
+    udw_course_ids = course_df['warehouse_id'].to_list()
+    enrollment_df = pull_enrollment_data_from_udw(udw_course_ids)
+
+    # Gather user data
+    udw_user_ids = enrollment_df['user_id'].drop_duplicates().to_list()
+    user_df = pull_user_data_from_udw(udw_user_ids)
+
+    # Find and remove rows with nonexistent user ids from enrollment_df
+    # This can take a few minutes
+    logger.info('Looking for rows with nonexistent user ids in enrollment data')
+    valid_user_ids = user_df['warehouse_id'].drop_duplicates().to_list()
+
+    enrollment_df['valid_id'] = enrollment_df['user_id'].map(
+        lambda x: check_if_valid_user_id(x, valid_user_ids)
+    )
+    enrollment_df = enrollment_df[(enrollment_df['valid_id'])].drop(columns=['valid_id'])
+
+    if CREATE_CSVS:
+        # Generate CSV Output
+        course_df.to_csv(os.path.join('data', 'course.csv'), index=False)
+        logger.info('Course data was written to data/course.csv')
+        user_df.to_csv(os.path.join('data', 'user_df.csv'), index=False)
+        logger.info('User data was written to data/user.csv')
+        enrollment_df.to_csv(os.path.join('data', 'enrollment.csv'), index=False)
+        logger.info('Enrollment data was written to data/enrollment.csv')
+
+    # Reset database
+    db_creator_obj = DBCreator('course_inventory', tables)
+    db_creator_obj.set_up_database()
+
+    # Insert gathered data
+    logger.info('Inserting course data')
+    course_df.to_sql('course', MYSQL_ENGINE, if_exists='append', index=False)
+    logger.info('Inserting user data')
```

Comment:
Can we put the dataframe size information into the log message? Same for the csv output logs. Thanks!

Reply:
This should be taken care of by [this commit](https://github.com/tl-its-umich-edu/course-inventory/pull/10/commits/0332accecb4912348fbc427fef75f2cfbe153c19).